src/modules/user-auth/contexts/AuthContext.tsx

```diff
@@ -7,6 +7,7 @@ import { createContext, useContext, FC } from 'react';
 import AuthService from '@services/AuthService';
 
 type AuthContextValue = {
+  ready: boolean;
```

Comment:
May need to add documentation or a different name. Really vague what "ready" can mean. And I don't understand what it does.

Reply:
Firebase client SDK validates the session with their server when the app starts. Before that, `user` is always null even if the client has a token in `localStorage`. `ready` means that the fetch is completed. Any name suggestion?

Reply:
So this is used to accompany the reusability of AuthGuard for both component and page level?

Reply:
Kinda but it has a bigger use case than that. Look at `AuthGuard.tsx, line 25`,  without the ready, the dialog will show up anyway if the user navigate directly to `/account` because the `user` object is always null on first load

Reply:
Chnage to `isAuthReady` for clarity

Reply:
>  Before that, `user` is always `null` even if the client has a token in `localStorage`.

Does initing `user` as undefined solve the problem?

Reply:
Maybe but then we will have to differentiating `user` between `undefined` which means not ready, `not` which means not logged in and the real value. This sounds like an agly solution to me...

Reply:
if seems intuitive to me, undefined means... undefined while null means not there. The reason I'm insisting on this is that `isAuthReady` seems like a derived value from `user`. That means there will be edge cases where we forget to sync these two.

Reply:
All the syncing we need is already there, we don't need to worry about it anymore in the future. The `user` is only unavailable for a split second when the app starts.

If we allow the `user` to be `undefined`, the check in `AuthGuard.tsx, line 25` will become 
```jsx
if (promptLogin && user === null) {
    ...
```
 It is really easy to forget why we do explicit `null` comparison there and at some point someone will try to "clean" it by making it
```jsx
if (promptLogin && !user) {
    ...
```
and the app breaks